bundle/src/main/java/com/adobe/acs/tools/explain_query/impl/ExplainQueryServlet.java

```diff
@@ -248,15 +259,17 @@ private JSONObject explainQuery(final Session session, final String statement,
             final Query query = queryManager.createQuery("explain " + effectiveStatement, effectiveLanguage);
             queryResult = query.execute();
         } finally {
-            if (logCollector != null) {
-                List<String> logs = logCollector.getLogs(collectorKey);
-                json.put("logs", logCollector.getLogs(collectorKey));
-
-                if (logs.size() == logCollector.msgCountLimit){
-                    json.put("logsTruncated", true);
+            synchronized (this.logCollector) {
```

Comment:
Same as before, but added synchronized as logCollector is a shared object now; unlikely to be problematic, but why not make sure.